pkg/relayer/session/sessiontree.go

```diff
@@ -95,7 +96,13 @@ func NewSessionTree(
 	// contain a non-hashed Relay that could be used to validate the proof on-chain.
 	trie := smt.NewSparseMerkleSumTrie(treeStore, protocol.NewTrieHasher(), smt.WithValueHasher(nil))
 
+	logger := polylog.Ctx(context.TODO()).With(
```

Comment:
`logger` should be passed to `NewSessionTree`.

We should keep the `With` method though.